qt/python/mantidqt/widgets/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+include ( SipQtTargetFunctions )
+
+set ( COMMON_INC_DIR ../../../widgets/common/inc )
+set ( HEADER_DEPENDS
+  ${COMMON_INC_DIR}/MantidQtWidgets/Common/AlgorithmDialog.h
+  ${COMMON_INC_DIR}/MantidQtWidgets/Common/AlgorithmDialog.h
+  ${COMMON_INC_DIR}/MantidQtWidgets/Common/Message.h
+  ${COMMON_INC_DIR}/MantidQtWidgets/Common/MessageDisplay.h
+)
+
+find_package ( Boost COMPONENTS python REQUIRED )
```

Comment:
I think the failure of the Python 3 build is due to this - it's find boost linked to the wrong version of Python. See [PythonInterface/CMakeLists.txt(mantidproject/mantid/blob/master/Framework/PythonInterface/CMakeLists.txt#L19) here for what happens in the framework. Perhaps we should add a `FindBoostPython.cmake` file to `buildconfig/CMake` to put this in a common place? It can then be called as `find_package ( BoostPython REQUIRED )`?